src/commands/appSettings/SqlDatabaseConnectionCustomPromptStep.ts

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import { ConnectionType } from '../../constants';
+import { getInvalidLengthMessage } from '../../constants-nls';
+import { localize } from '../../localize';
+import { validateUtils } from '../../utils/validateUtils';
+import { ISqlDatabaseConnectionWizardContext } from './ISqlDatabaseConnectionWizardContext';
+
+export class SqlDatabaseConnectionCustomPromptStep<T extends ISqlDatabaseConnectionWizardContext> extends AzureWizardPromptStep<T> {
+    public async prompt(context: T): Promise<void> {
+        context.nonAzureSqlConnection = (await context.ui.showInputBox({
+            prompt: localize('customSqlConnectionPrompt', 'Enter a custom SQL connection string.'),
+            validateInput: (value: string | undefined) => this._validateInput(value)
+        })).trim();
+    }
+
+    public shouldPrompt(context: T): boolean {
+        // 'NonAzure' represents any local or remote custom SQL connection that is not hosted through Azure
```

Comment:
I like having this comment, however, I think it'd be best to add it as a doc comment to the actual enum value instead. That way you can see it on hover of `ConnectionType.NonAzure` anywhere it's used.

Reply:
Will do.  In case there are still other comments out there like this one, I am planning to change this pattern in a near-future PR to address this issue: https://github.com/microsoft/vscode-azurefunctions/issues/3448 

During that time, I will be revisiting and changing these sections to reflect the new `ConnectionTypes`.